src/android/nl/xservices/plugins/Calendar.java

```diff
@@ -213,7 +213,8 @@ public void onRequestPermissionResult(int requestCode, String[] permissions, int
 
   private void openCalendarLegacy(JSONArray args) {
     try {
-      final Long millis = args.getJSONObject(0).optLong("date");
+      long provided = args.getJSONObject(0).optLong("date", -1);
+      final long millis = provided < 0 ? new Date().getTime() : provided;
```

Comment:
In this case the provided date has a negative millis value,
which makes the app crash when handling the intent.

This change makes any negative value default to now.
Zero value is still handled as zero.